webworker/index.ts

```diff
@@ -10,7 +10,32 @@ import * as React from "react"
 import * as jscadFiber from "jscad-fiber"
 import { getImportsFromCode } from "@tscircuit/prompt-benchmarks/code-runner-utils"
 
+class WebWorkerEventEmitter {
+  private listeners: Record<string, Array<(...args: any[]) => void>> = {}
+
+  on(event: string, callback: (...args: any[]) => void) {
+    if (!this.listeners[event]) {
+      this.listeners[event] = []
+    }
+    this.listeners[event].push(callback)
+  }
+
+  emit(event: string, ...args: any[]) {
+    if (!this.listeners[event]) return
+    this.listeners[event].forEach((listener) => {
+      try {
+        listener(...args)
+      } catch (error) {
+        console.error(`Error in event listener for ${event}:`, error)
+      }
+    })
+  }
+}
+
+const globalEventEmitter = new WebWorkerEventEmitter()
```

Comment:
if this is just a normal event emitted (it looks like it is) you can just initialize `EventEmitter` from node

Reply:
also does this have to be global? Why not just initialize an event emitter inside the `create*` method?